src/AIAPI/MainGame/GameAPI.Hooks.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+// using ActionGame;
+using HarmonyLib;
+using AIProject.SaveData;
+using Manager;
+using AIProject;
+using Sirenix.Serialization;
+
+namespace KKAPI.MainGame
+{
+    public static partial class GameAPI
+    {
+        private class Hooks
+        {
+            public static int lastCurrentDay = 1;//Always starts at 1
+
+            public static void SetupHooks()
+            {
+                Harmony.CreateAndPatchAll(typeof(Hooks));
+            }
+
+            //TODO SaveData.Load only has 1 param in AI and it's the path, does not include filename like KK
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(SaveData), nameof(SaveData.Load), new[] { typeof(string) })]
+            public static void LoadHook(string fileName)
+            {
+                OnGameBeingLoaded("", fileName);
+            }
+
+            [HarmonyPrefix]
+            [HarmonyPatch(typeof(SaveData), nameof(SaveData.SaveFile), new[] { typeof(string) })]
+            public static void SaveHook(string path)
+            {
+                OnGameBeingSaved(path, "");
+            }
+
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(HScene), "InitCoroutine")]
+            public static void HScene_InitCoroutine(HScene __instance)
```

Comment:
This Hook is not working just yet.  Can't seem to find a good HScene.Start hook.

Reply:
BetterHScenes uses SetStartVoice... It gets called near the end of HScene Start... A little wonky but it seems to work

Reply:
This worked great once I realised the KK OnHScene copied method was an IEnumerator...  I had even tried SetStartVoice earlier and thought it didn't work.   Thanks for the reminder!